storybook/test-runner.ts

```diff
@@ -0,0 +1,31 @@
+import { getStoryContext, TestHook } from "@storybook/test-runner"
+import { configureAxe, getAxeResults, injectAxe } from "axe-playwright"
+import { toHaveNoViolations } from "jest-axe"
+
+export const setup: () => void = () => {
+  expect.extend(toHaveNoViolations)
+}
+
+export const preRender: TestHook = async page => {
+  await injectAxe(page)
+}
+
+export const postRender: TestHook = async (page, context) => {
+  const { parameters } = await getStoryContext(page, context)
+
+  if (parameters?.a11y?.disable) {
+    return
+  }
+
+  await configureAxe(page, parameters?.a11y?.config ?? {})
```

Comment:
You could potentially put the global rules somewhere in this function and have it combine with any that you retrieve from the story context (custom rules). That way you should be able to not need to import the global ones in your individual stories where you're trying to extend rules :)

Reply:
Great idea, I've implemented that now